### PR TITLE
Chat window implementation with all functions working

### PR DESCRIPTION
**Pull request by ravisumit33**

### CODE REVIEW

File: frontend/src/components/ChatContainer/ChatWindow/ChatHeader/index.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+}));
+
+const ChatHeader = (props) => {
+  const { chatId } = props;
+  const classes = useStyles();
+  const chatWindowStore = useContext(ChatWindowStoreContext);
+  const { name, setWindowMinimized, reconnect } = chatWindowStore;
+
+  const handleMinimize = () => {
+    setWindowMinimized(true);
```
Comment (nileshvaishnav) [inline frontend/src/components/ChatContainer/ChatWindow/ChatHeader/index.js:43]: 'this' is undefined when calling this funtion using object destructuring
but works fine if called on chatWidnowStore(chatWindowStore.setWindowMinimized(ture)).

Please check if 'this' is bound or not for setWindowMinimized.
Also even if it is bound using action.bound, are we allowed to use object destructuring and call the function?

**Review by nileshvaishnav — CHANGES_REQUESTED** [review]
Please look into commented issue

File: frontend/src/components/ChatContainer/ChatWindow/ChatHeader/index.js (hunk 1 above)
Comment (ravisumit33) [reply frontend/src/components/ChatContainer/ChatWindow/ChatHeader/index.js:43]: Classes are just objects in javascript so we can freely destructure.

**Review by nileshvaishnav — APPROVED** [review]
LGTM

**Pull request merged**